charmcraft/templates/init-kubernetes/tests/spread/lib/tools/retry.j2

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+import argparse
+import itertools
+import os
+import subprocess
+import sys
+import time
+
+
+def envpair(s: str) -> str:
+    if not "=" in s:
+        raise argparse.ArgumentTypeError(
+            "environment variables expected format is 'KEY=VAL' got '{}'".format(s)
+        )
+    return s
+
+
+def _make_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description="""
+Retry executes COMMAND at most N times, waiting for SECONDS between each
+attempt. On failure the exit code from the final attempt is returned.
+"""
+    )
+    parser.add_argument(
+        "-n",
+        "--attempts",
+        metavar="N",
+        type=int,
+        default=10,
+        help="number of attempts (default %(default)s)",
+    )
+    parser.add_argument(
+        "--wait",
+        metavar="SECONDS",
+        type=float,
+        default=5,
+        help="grace period between attempts (default %(default)ss)",
+    )
+    parser.add_argument(
+        "--env",
+        type=envpair,
+        metavar="KEY=VAL",
+        action="append",
+        default=[],
+        help="environment variable to use with format KEY=VALUE (no default)",
+    )
+    parser.add_argument(
+        "--maxmins",
```

Comment:
I thought this was 'maxims' at first :smile: 

This is easier to read and follows the same style as `--expect-rc`.
```suggestion
        "--max-mins",
```

Reply:
This file was copied as-is from the parca tests, fixing as suggested.